src/main/kotlin/com/thelastpickle/tlpstress/profiles/BasicTimeSeries.kt

```diff
@@ -29,16 +32,28 @@ class BasicTimeSeries : IStressProfile {
     }
 
     lateinit var prepared: PreparedStatement
-    lateinit var getRow: PreparedStatement
     lateinit var getPartitionHead: PreparedStatement
+    lateinit var delete: PreparedStatement
+    lateinit var deletePartitionHead: PreparedStatement
+    lateinit var cassandraVersion: VersionNumber
 
     @WorkloadParameter("Number of rows to fetch back on SELECT queries")
     var limit = 500
 
+    @WorkloadParameter("Deletion range in seconds. Range tombstones will cover all rows older than the given value.")
+    var deleteDepth = 30
+
     override fun prepare(session: Session) {
+        println("Using a limit of $limit for reads and deleting data older than $deleteDepth seconds (if enabled).")
+        cassandraVersion = session.cluster.metadata.allHosts.map { host -> host.cassandraVersion }.min()!!
         prepared = session.prepare("INSERT INTO sensor_data (sensor_id, timestamp, data) VALUES (?, ?, ?)")
-        getRow = session.prepare("SELECT * from sensor_data WHERE sensor_id = ? AND timestamp = ? ")
         getPartitionHead = session.prepare("SELECT * from sensor_data WHERE sensor_id = ? LIMIT ?")
+        if (cassandraVersion.compareTo(VersionNumber.parse("3.0")) >= 0) {
```

Comment:
This is a bit weird.  I think we should allow `--deletes` on versions >= 3.0 only, and error if it's supplied with 2.x.  Behaving differently between the versions is going to lead to inaccurate benchmarks.

Reply:
Sounds legit, I now throw an exception when trying to use `--deletes` with the BasicTimeSeries workload if Cassandra < 3.0.